### PR TITLE
Build examples on CMake

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Comment by saghul** [comment]
@bnoordhuis This is ready now! The only missing thing in order to be able to drop the Makefile is cross-compilation for MinGW, but I haven't been able to figure it all out yet. Are you OK having that later? If so I'll go ahead and rewrite the Makefile tomorrow.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,6 +48,31 @@ jobs:
...
+          ldd build/test_fib
+          ./build/hello
+          ./build/hello_module
+          ./build/test_fib
+          cp build/fib.so examples/
+          ./build/qjs examples/test_fib.js
+          cp build/point.so examples/
+          ./build/qjs examples/test_point.js
+          cp build/bjson.so tests/
+          ./build/qjs tests/test_bjson.js
```
Comment (bnoordhuis) [inline .github/workflows/ci.yml]: The most minor of nits but IMO it's a little easier to read when written like this (edit: the diff itself isn't all that legible, I'm afraid):
```suggestion
          cp build/fib.so examples/
          cp build/point.so examples/
          cp build/bjson.so tests/
          ./build/qjs examples/test_fib.js
          ./build/qjs examples/test_point.js
          ./build/qjs tests/test_bjson.js
```
Alternatively, you could use LD_LIBRARY_PATH/DYLD_LIBRARY_PATH so you don't have to copy the files.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -175,6 +200,31 @@ jobs:
...
+          otool -L build/test_fib
+          ./build/hello
+          ./build/hello_module
+          ./build/test_fib
+          cp build/fib.so examples/
+          ./build/qjs examples/test_fib.js
+          cp build/point.so examples/
+          ./build/qjs examples/test_point.js
+          cp build/bjson.so tests/
+          ./build/qjs tests/test_bjson.js
```
Comment (bnoordhuis) [inline .github/workflows/ci.yml]: Likewise:
```suggestion
          cp build/fib.so examples/
          cp build/point.so examples/
          cp build/bjson.so tests/
          ./build/qjs examples/test_fib.js
          ./build/qjs examples/test_point.js
          ./build/qjs tests/test_bjson.js
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -174,9 +176,114 @@ add_executable(run-test262
...
+# Examples
+#
+
+if(BUILD_EXAMPLES AND NOT MINGW)
+    list(APPEND HELLO_OPTS
+        -fno-string-normalize
+        -fno-map
+        -fno-promise
+        -fno-typedarray
+        -fno-typedarray
```
Comment (bnoordhuis) [inline CMakeLists.txt:191]:
```suggestion
        -fno-typedarray
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -174,9 +176,114 @@ add_executable(run-test262
...
+    )
+    target_include_directories(hello PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+    target_compile_definitions(hello PRIVATE ${qjs_defines})
+    target_link_libraries(hello ${qjs_libs})
+
+    list(APPEND HELLO_MODULE_OPTS
+        -fno-string-normalize
+        -fno-map
+        -fno-typedarray
+        -fno-typedarray
```
Comment (bnoordhuis) [inline CMakeLists.txt]:
```suggestion
        -fno-typedarray
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -174,9 +176,114 @@ add_executable(run-test262
...
+    )
+    target_include_directories(hello_module PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+    target_compile_definitions(hello_module PRIVATE ${qjs_defines})
+    target_link_libraries(hello_module ${qjs_libs})
+
+    if(NOT MINGW)
+        add_library(fib MODULE examples/fib.c)
+        set_target_properties(fib PROPERTIES
+            PREFIX ""
+        )
```
Comment (bnoordhuis) [inline CMakeLists.txt:245]: For my understanding: that strips the "lib" prefix, right?

**Review by bnoordhuis — APPROVED** [review]
:+1:

File: CMakeLists.txt (hunk 5 above)
Comment (saghul) [reply CMakeLists.txt:245]: Correct!

**Pull request merged**